extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -62,6 +65,22 @@ export class AmpInstallServiceWorker extends AMP.BaseElement {
     assertHttpsUrl(src, this.element);
 
     if (isProxyOrigin(src) || isProxyOrigin(win.location.href)) {
+      // Schedule loading the iframe in firstLayoutCompleted().
+      this.shouldInsertframeOnFirstLayout_ = true;
+    } else if (parseUrl(win.location.href).origin == parseUrl(src).origin) {
+      this.loadPromise(this.win).then(() => {
+        return install(this.win, src);
+      });
+    } else {
+      this.user().error(TAG,
+          'Did not install ServiceWorker because it does not ' +
+          'match the current origin: ' + src);
+    }
+  }
+
+  /** @override */
+  firstLayoutCompleted() {
```

Comment:
Use `layoutCallback`. We'll need to guard against layout happening, regardless.

Reply:
Done